src/main/java/org/opensearch/security/user/UserTokenHandler.java

```diff
@@ -0,0 +1,184 @@
+
+    Client client;
+
+    final static String FAILED_CLEAR_HASH_MESSAGE = "The hash could not be cleared from the specified account.";
+
+    public static CType getRevokedTokensConfigName() {
+        return CType.REVOKEDTOKENS;
+    }
+
+    @Inject
+    public UserTokenHandler(
+        ThreadPool threadPool,
+        ClusterService clusterService,
+        ConfigurationRepository configurationRepository,
+        Client client
+    ) {
+        this.settings = Settings.builder().put("signing_key", signingKey).put("encryption_key", claimsEncryptionKey).build();
+        this.jwtVendor = new JwtVendor(settings, null);
+        this.threadPool = threadPool;
+        this.clusterService = clusterService;
+        this.client = client;
+        this.configurationRepository = configurationRepository;
+    }
+
+    // public AuthToken issueToken(String audience) {
+    // ThreadContext threadContext = threadPool.getThreadContext();
+    // final User user = threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER);
+    // String jwt = null;
+    // try {
+    // jwt = jwtVendor.createJwt(
+    // clusterService.getClusterName().toString(),
+    // user.getName(),
+    // audience,
+    // DEFAULT_EXPIRATION_TIME_SECONDS,
+    // new ArrayList<String>(user.getRoles()),
+    // null
+    // );
+    // } catch (Exception e) {
+    // throw new RuntimeException(e);
+    // }
+    // return new BearerAuthToken(jwt);
+    // }
+
+    public boolean validateToken(AuthToken authToken) {
+        if (!(authToken instanceof BearerAuthToken)) {
+            throw new UserServiceException("The provided token is not a BearerAuthToken.");
+        }
+        BearerAuthToken bearerAuthToken = (BearerAuthToken) authToken;
+        JwsJwtCompactConsumer jwtConsumer = new JwsJwtCompactConsumer(bearerAuthToken.getCompleteToken());
+        JwtToken jwt = jwtConsumer.getJwtToken();
+
+        Long iat = (Long) jwt.getClaim("iat");
+        Long exp = (Long) jwt.getClaim("exp");
+        SecurityDynamicConfiguration revokedTokens = load(getRevokedTokensConfigName(), false);
+        Long currentTime = System.currentTimeMillis();
+        return (exp > currentTime && !revokedTokens.exists(bearerAuthToken.getCompleteToken()));
+    }
+
+    public boolean validateJustInTimeToken(AuthToken authToken) {
+        if (!(authToken instanceof BearerAuthToken)) {
+            throw new UserServiceException("The provided token is not a BearerAuthToken.");
+        }
+        BearerAuthToken bearerAuthToken = (BearerAuthToken) authToken;
+        JwsJwtCompactConsumer jwtConsumer = new JwsJwtCompactConsumer(bearerAuthToken.getCompleteToken());
+        JwtToken jwt = jwtConsumer.getJwtToken();
+
+        Long iat = (Long) jwt.getClaim("iat");
+        Long nbf = (Long) jwt.getClaim("nbf");
+        Long exp = (Long) jwt.getClaim("exp");
+        SecurityDynamicConfiguration revokedTokens = load(getRevokedTokensConfigName(), false);
```

Comment:
If we choose to split the validations between obo tokens and other bearer auth tokens in this way, then I don't think we need this line here, since the revocation is not in the scope of obo token. (At least for now)